src/ethereum/vm/instructions.py

```diff
@@ -88,27 +404,72 @@ def sstore(evm: Evm) -> None:
         evm.refund_counter += GAS_STORAGE_CLEAR_REFUND
 
     if new_value == 0:
-        del evm.env.state[evm.current].storage[key]
+        # Deletes a k-v pair from dict if key is present, else does nothing
+        evm.env.state[evm.current].storage.pop(key, None)
     else:
         evm.env.state[evm.current].storage[key] = new_value
 
 
-def push1(evm: Evm) -> None:
+def pushN(evm: Evm, num_bytes: int) -> None:
     """
-    Pushes a one-byte immediate onto the stack.
+    Pushes a N-byte immediate onto the stack.
 
     Parameters
     ----------
-    evm :
+    evm : `Evm`
         The current EVM frame.
 
+    num_bytes : `int`
+        The number of immediate bytes to be read from the code and pushed to
+        the stack.
+
     Raises
     ------
     StackOverflowError
         If `len(stack)` is equals `1024`.
     OutOfGasError
         If `evm.gas_left` is less than `GAS_VERY_LOW`.
     """
+    assert evm.pc + num_bytes < len(evm.code)
     evm.gas_left = subtract_gas(evm.gas_left, GAS_VERY_LOW)
-    push(evm.stack, U256(evm.code[evm.pc + 1]))
-    evm.pc += 1
+
+    data_to_push = U256.from_be_bytes(
+        evm.code[evm.pc + 1 : evm.pc + num_bytes + 1]
+    )
+    push(evm.stack, data_to_push)
+
+    evm.pc += num_bytes
+
+
+push1 = partial(pushN, num_bytes=1)
```

Comment:
I'm somewhat torn on this.

I don't think copying and pasting the push implementation 32 times would actually add any value, but it would be the most obvious implementation...

Reply:
I'm also rather torn on this. I think we're doing a good job keeping things really clear, maybe we do take a small short cut here?

Reply:
I vote for `partial` for the time being.